app/transactions/ams/steps/create_aapb_admin_data.rb

```diff
@@ -51,22 +51,26 @@ def save_aapb_admin_data(change_set)
       end
 
       def set_admin_data_attributes(admin_data, change_set)
-        AdminData.attributes_for_update.each do |k|
-          # Some attributes are serialized on AdminData, so always send an array
-          k_string = k.to_s
-          if should_empty_admin_data_value?(k, change_set)
-            AdminData::SERIALIZED_FIELDS.include?(k) ? admin_data.send("#{k}=", Array.new) : admin_data.send("#{k}=", nil)
-          elsif change_set.fields[k_string].present?
-            AdminData::SERIALIZED_FIELDS.include?(k) ? admin_data.send("#{k}=", Array(change_set.fields[k_string])) : admin_data.send("#{k}=", change_set.fields[k_string].to_s)
+        AdminData.attributes_for_update.each do |field|
+          field = field.to_s
+          # Convert emtpy strings to nil, whether the value is a scalar or an array
+          new_admin_data_value = if change_set.fields[field].respond_to?(:select)
+            change_set.fields[field].reject {|v| v.blank? }
+          elsif change_set.fields[field].blank?
+            nil
+          end
```

Comment:
Can you confirm if I'm reading this conditional correctly? 

```
if can be multiple
  filter out blank values
else if single and blank
  set new value to nil
end
```

If I am reading that correctly, I think we're missing a scenario, which would be "is single and **not** blank". In that case, I think we just lose the value. I think something like this would resolve this: 

```suggestion
          new_admin_data_value = if change_set.fields[field].respond_to?(:select)
            change_set.fields[field].reject {|v| v.blank? }
          elsif change_set.fields[field].blank?
            nil
          else # non-multiple with value present
            change_set.fields[field]
          end
```